pandas/tests/window/moments/test_moments_expanding.py

```diff
@@ -0,0 +1,318 @@
+        )
+        # TODO: xref gh-15826
+        # .loc is not preserving the names
+        result1 = df1.expanding().cov(df2, pairwise=True).loc[2]
+        result2 = df1.expanding().cov(df2a, pairwise=True).loc[2]
+        result3 = df1a.expanding().cov(df2, pairwise=True).loc[2]
+        result4 = df1a.expanding().cov(df2a, pairwise=True).loc[2]
+        expected = DataFrame(
+            [[-3.0, -6.0], [-5.0, -10.0]],
+            columns=Index(["A", "B"], name="foo"),
+            index=Index(["X", "Y"], name="foo"),
+        )
+        tm.assert_frame_equal(result1, expected)
+        tm.assert_frame_equal(result2, expected)
+        tm.assert_frame_equal(result3, expected)
+        tm.assert_frame_equal(result4, expected)
+
+    def test_expanding_corr_pairwise_diff_length(self):
+        # GH 7512
+        df1 = DataFrame(
+            [[1, 2], [3, 2], [3, 4]],
+            columns=["A", "B"],
+            index=Index(range(3), name="bar"),
+        )
+        df1a = DataFrame(
+            [[1, 2], [3, 4]], index=Index([0, 2], name="bar"), columns=["A", "B"]
+        )
+        df2 = DataFrame(
+            [[5, 6], [None, None], [2, 1]],
+            columns=["X", "Y"],
+            index=Index(range(3), name="bar"),
+        )
+        df2a = DataFrame(
+            [[5, 6], [2, 1]], index=Index([0, 2], name="bar"), columns=["X", "Y"]
+        )
+        result1 = df1.expanding().corr(df2, pairwise=True).loc[2]
+        result2 = df1.expanding().corr(df2a, pairwise=True).loc[2]
+        result3 = df1a.expanding().corr(df2, pairwise=True).loc[2]
+        result4 = df1a.expanding().corr(df2a, pairwise=True).loc[2]
+        expected = DataFrame(
+            [[-1.0, -1.0], [-1.0, -1.0]], columns=["A", "B"], index=Index(["X", "Y"])
+        )
+        tm.assert_frame_equal(result1, expected)
+        tm.assert_frame_equal(result2, expected)
+        tm.assert_frame_equal(result3, expected)
+        tm.assert_frame_equal(result4, expected)
+
+    @pytest.mark.parametrize(
+        "func,static_comp",
+        [("sum", np.sum), ("mean", np.mean), ("max", np.max), ("min", np.min)],
+        ids=["sum", "mean", "max", "min"],
+    )
+    def test_expanding_func(self, func, static_comp):
+        def expanding_func(x, min_periods=1, center=False, axis=0):
+            exp = x.expanding(min_periods=min_periods, center=center, axis=axis)
+            return getattr(exp, func)()
+
+        self._check_expanding(expanding_func, static_comp, preserve_nan=False)
+
+    def test_expanding_apply(self, raw):
+        def expanding_mean(x, min_periods=1):
+
+            exp = x.expanding(min_periods=min_periods)
+            result = exp.apply(lambda x: x.mean(), raw=raw)
+            return result
+
+        # TODO(jreback), needed to add preserve_nan=False
+        # here to make this pass
+        self._check_expanding(expanding_mean, np.mean, preserve_nan=False)
+
+        ser = Series([], dtype=np.float64)
```

Comment:
This test looks like it could stand alone in its own function.

Reply:
sefinitely